From 7eaeaa34de1223c0fe0b66cb1adfab452d540421 Mon Sep 17 00:00:00 2001
From: Karl Dietz <dekarl@mythtv.org>
Date: Wed, 8 Jan 2014 08:05:21 +0100
Subject: [PATCH 51/57] change IPTV Url test to consider RTSP without explicit
 port as valid

Found by Nicolas Niclausse
Refs #11949

(cherry picked from commit a93e14d3f528328e8e280c54003fb2a4bc4d47aa)

Conflicts:
	mythtv/libs/libmythtv/test/test_iptvrecorder/test_iptvrecorder.h
---
 mythtv/libs/libmythtv/iptvtuningdata.h |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/mythtv/libs/libmythtv/iptvtuningdata.h b/mythtv/libs/libmythtv/iptvtuningdata.h
index 0ebe2ff..471f54f 100644
--- a/mythtv/libs/libmythtv/iptvtuningdata.h
+++ b/mythtv/libs/libmythtv/iptvtuningdata.h
@@ -140,7 +140,11 @@ class MTV_PUBLIC IPTVTuningData
 
     bool IsValid(void) const
     {
-        return IsHLS() || m_data_url.port() != -1;
+        return m_data_url.isValid() &&
+               (IsHLS() ||
+                ((m_data_url.scheme() == "rtp" || m_data_url.scheme() == "udp") && m_data_url.port() != -1) ||
+                m_data_url.scheme() == "rtsp"
+               );
     }
 
   public:
-- 
1.7.10.2

